CoreFoundation/Base.subproj/CFInternal.h

```diff
@@ -412,11 +412,11 @@ CF_EXPORT void *__CFConstantStringClassReference[];
 #endif
 
 #define CONST_STRING_DECL(S, V) \
-const struct __CFConstStr __##S CONST_STRING_SECTION = {{(uintptr_t)&__CFConstantStringClassReference, _CF_CONSTANT_OBJECT_STRONG_RC, 0, 0x000007c8U}, (uint8_t *)(V), sizeof(V) - 1}; \
+const struct __CFConstStr __##S CONST_STRING_SECTION = {{(uintptr_t)&__CFConstantStringClassReference, _CF_CONSTANT_OBJECT_STRONG_RC, 0x000007c8U}, (uint8_t *)(V), sizeof(V) - 1}; \
```

Comment:
Do you need to change the definition of `_CF_CONSTANT_OBJECT_STRONG_RC`? My guess is that the bits generated here will be wrong if you do not.

Reply:
This will work on little-endian only.. are there are any big-endian Swift ports?

Reply:
I think the point of splitting it into strong/weak here was so we could set that pinned flag. There may now be a better way.

Reply:
The Android tests run through ok. Are you saying there is still something to change?

Reply:
I don't think you were setting the pinned flag correctly before. The pinned flag is the LSB of that `uintptr_t` on all platforms.

I don't think you are setting the refcount correctly now. The strong refcount is stored differently on 32-bit versus 64-bit, but on all platforms it is biased: a strong refcount of 1 is represented as bit value 0.

Reply:
#define _CF_CONSTANT_OBJECT_STRONG_RC (_CF_SWIFT_RC_PINNED_FLAG) then?